packages/bitcoin/src/bitcoin.utils.ts

```diff
@@ -89,32 +89,33 @@ export function decodeBitcoinTx(tx: string): ReturnType<typeof btc.RawTx.decode>
 export function getAddressFromOutScript(script: Uint8Array, bitcoinNetwork: BtcSignerNetwork) {
   const outputScript = btc.OutScript.decode(script);
 
-  if (outputScript.type === 'pk' || outputScript.type === 'tr') {
-    return btc.Address(bitcoinNetwork).encode({
-      type: outputScript.type,
-      pubkey: outputScript.pubkey,
-    });
+  switch (outputScript.type) {
+    case 'pkh':
+    case 'sh':
+    case 'wpkh':
+    case 'wsh':
+      return btc.Address(bitcoinNetwork).encode({
+        type: outputScript.type,
+        hash: outputScript.hash,
+      });
+    case 'tr':
+      return btc.Address(bitcoinNetwork).encode({
+        type: outputScript.type,
+        pubkey: outputScript.pubkey,
+      });
+    case 'ms':
+      return btc.p2ms(outputScript.m, outputScript.pubkeys).address ?? '';
+    case 'pk':
+      return btc.p2pk(outputScript.pubkey, bitcoinNetwork).address ?? '';
+    case 'tr_ms':
```

Comment:
I think we can get rid of `logger.error(`Cannot currently handle script type ${outputScript.type}`)` and catch `tr_ns` and `tr_ms` with the `default` case

Reply:
I just made that change, hopefully that's OK with you @fbwoolf ?